### PR TITLE
Show group validation errors

### LINKED ISSUE #21: Group name validation doesn't show an error message
Intended behavior: Error message in form

Actual behavior: A redirect to the same form with all of the fields cleared out and no error message.

How to reproduce; Attempt to create a group with the name `Illegal$Name` under any center and competition.

### PR DESCRIPTION
**Pull request by Yotam7**
closes  #21
image
image
image
image

### CODE REVIEW

File: codeguru/views.py [hunk 1]
```diff
@@ -54,6 +51,16 @@ def group(request, id=None):
 
                 return redirect("group", id=new_group.id)
 
+            if CgGroup.objects.all().filter(name=form.data["name"]).exists():
+                return error(request, gettext("Group already exists. try choosing another name."))
```
Comment (rootatkali) [inline codeguru/views.py:55]: This check was moved from before the group's creation to outside `is_valid`. Does it still work as expected?

File: codeguru/views.py [hunk 2]
```diff
@@ -54,6 +51,16 @@ def group(request, id=None):
...
 
+            if CgGroup.objects.all().filter(name=form.data["name"]).exists():
+                return error(request, gettext("Group already exists. try choosing another name."))
+            else:
+                return error(
+                    request,
+                    gettext(
+                        "Group name is invalid. Groups names may only include alphanumerical characters and underscores"
+                    ),
+                )
```
Comment (rootatkali) [inline codeguru/views.py:62]: In theory, we can also use `Form.errors` to get the exact error message the form raised. Not critical if it's a lot of work or complicates things.

Also, what if the submitted form has a different error? (for example - no center)

File: codeguru/views.py (hunk 1 above)
Comment (Yotam7) [reply codeguru/views.py:55]: This move was intended, `CgGroup`'s model has a check for the uniqueness of the name, thus if the name isn't unique the form won't be valid
https://github.com/codeguru-il/codeguru-website/blob/0fee6aab344081ec5b16ab99ff33b6e07db39c97/codeguru/models.py#L65

File: codeguru/views.py (hunk 2 above)
Comment (Yotam7) [reply codeguru/views.py:64]: tried using `form.errors`  but I could not figure out how to get it to translate "name"
image
added case for invalid name, if the error is not invalid/taken name, it returns `form.errors` (which probably won't be fully translated)

**Review by rootatkali — APPROVED** [review]
As long as the recent changes work, this PR is approved :)

**Pull request merged**